extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/AbstractRestClientConfigBuilder.java

```diff
@@ -25,16 +33,69 @@ public abstract class AbstractRestClientConfigBuilder implements ConfigBuilder {
     @Override
     public SmallRyeConfigBuilder configBuilder(final SmallRyeConfigBuilder builder) {
         List<RegisteredRestClient> restClients = getRestClients();
-        builder.withInterceptors(new RestClientNameFallbackConfigSourceInterceptor(restClients));
+        Set<String> ignoreNames = getIgnoreNames();
+        builder.withInterceptors(new RestClientNameUnquotedFallbackInterceptor(restClients, ignoreNames));
+        builder.withInterceptors(new RestClientNameFallbackInterceptor(restClients, ignoreNames));
         for (RegisteredRestClient restClient : restClients) {
             builder.withDefaultValue("quarkus.rest-client.\"" + restClient.getFullName() + "\".force", "true");
             builder.withDefaultValue("quarkus.rest-client." + restClient.getSimpleName() + ".force", "true");
             if (restClient.getConfigKey() != null) {
-                builder.withDefaultValue("quarkus.rest-client." + restClient.getConfigKey() + ".force", "true");
+                builder.withDefaultValue("quarkus.rest-client.\"" + restClient.getConfigKey() + "\".force", "true");
             }
         }
```

Comment:
Why are we doing that and enforcing this value here? I don't understand why we would have to do that.

Reply:
OK, so I have done my homework about this thing and it seems reasonable.

I wonder though if we should come up with a way to filter it from the Dev UI? Maybe by filtering config that shouldn't appear in the doc? If it's not in the doc, I would thing that we wouldn't want it to be visible.

Or we would need another annotation.

Reply:
Note that I'm not requesting a change for this patch. Just thinking out loud for the future.

Reply:
I've tried to add that explanation here:
https://github.com/quarkusio/quarkus/blob/1da0004f06a0b17da0e25a71d3bb656c949aca7a/extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/AbstractRestClientConfigBuilder.java#L19-L27

Let me expand that further, and then I can update the javadoc too:

In config, when we populate a `Map`, we must rely on the list of property names because the key is dynamic. As an example, if we want to retrieve the list of datasources, we have to look for properties `quarkus.datasource."datasource-name"`, and then we do the rest. 

There is a catch: of course, this only works when we get at least one property of the tree with the expected name. The problem is that some sources (like Vault) cannot list the properties. In the example case, if all the datasource properties are only set in Vault, we cannot load that configuration because we don't know it exists. What we recommend is to set some base properties and then override them.

In the case of the REST Client, it's a bit of a mix. At the mapping level, we use a `Map` because, obviously, we can't represent an unknown number of clients in the mapping structure, but we do know all the possible keys in advance. So, I'm adding a dummy key to force the lookup. Once the Config knows the dynamic part of a key, it will then query all the other pieces directly so that the Vault config will work.

I plan to support this out of the box without having to rely on workarounds.

Reply:
Yeah, I missed it because of the context of the diff but the Javadoc you already have is perfectly clear. That's what I was trying to say in my previous message by "doing my homework" :)

Reply:
> I wonder though if we should come up with a way to filter it from the Dev UI? Maybe by filtering config that shouldn't appear in the doc? If it's not in the doc, I would thing that we wouldn't want it to be visible.

Yes, it is annoying, and I didn't think of that when I added it in the first place. We could filter it with the doc annotation, or maybe `@Deprecated`? 

Anyway, I'll add something in the Config system for such cases (we have the same case in OTel), where we can't statically map the configuration, but the keys are well-known. I'll add something that will always query specific keys without relying on the list provided by the sources, so these types of hacks will no longer be required.